chain/network/src/peer/peer_actor.rs

```diff
@@ -1396,8 +1383,31 @@ impl actix::Actor for PeerActor {
             }
             Some(reason) => {
                 tracing::info!(target: "network", "{:?}: Peer {} disconnected, reason: {reason}", self.my_node_info.id, self.peer_info);
+                let allow_reconnect = match reason {
+                    ClosingReason::TooManyInbound => false, // too many inbound
+                    ClosingReason::OutboundNotAllowed(_) => true,
```

Comment:
why for OutboundNotAllowed should allow for reconnect? Please document for each case (especially non-trivial ones) what is the reasoning behind allow_reconnect value.

Reply:
I mean, the ones that you documented LG, just please add more